samples/red_black_tree.cr

```diff
@@ -1,26 +1,28 @@
-# Copied with little modifications from: https://github.com/rubinius/rubinius-benchmark/blob/master/real_world/bench_red_black_tree.rb
+# Copied with little modifications from: https://github.com/rubinius/rubinius-benchmark/blob/cf4a2468f46d23cc300815afabc8150609383d6c/real_world/bench_red_black_tree.rb
 
 class RedBlackTree
   class Node
-    property :color
-    property :key
+    enum Color
+      Red
+      Black
+    end
+
+    property color : Color
+    property key : Int32
     property! :left
     property! :right
     property! parent : self
 
-    RED   = :red
-    BLACK = :black
-
-    def initialize(@key : Int32, @color = RED)
+    def initialize(@key, @color : Color = :red)
       @left = @right = @parent = NilNode.instance
     end
 
     def black?
-      color == BLACK
+      color.black?
     end
 
     def red?
-      color == RED
+      color.red?
     end
```

Comment:
This can be made more succinct

```suggestion
    delegate black?, red?, to: color
```

Reply:
True but I wanted to leave it for a more symmetrical comparison with the original.